oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2ClientAuthenticationProvider.java

```diff
@@ -64,11 +66,13 @@
 	 * @param authorizationService the authorization service
 	 */
 	public OAuth2ClientAuthenticationProvider(RegisteredClientRepository registeredClientRepository,
-			OAuth2AuthorizationService authorizationService) {
+			OAuth2AuthorizationService authorizationService, PasswordEncoder passwordEncoder) {
```

Comment:
Please revert constructor parameter and instead add `setPasswordEncoder()`. The default should be set using `PasswordEncoderFactories.createDelegatingPasswordEncoder()` and `setPasswordEncoder()` will only be called if a `@Bean` exists.

Reply:
Fixed.

Reply:
Hi @jgrandja. I'm marking this review comment as resolved, but I'm not sure how it works in Spring team. If reviewers should do it, let me know.